app/controllers/api_controller.rb

```diff
@@ -19,4 +20,12 @@ def validation_error(resource)
       ]
     }, status: :bad_request
   end
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    render json: { error: 'You are not authorized to perform this action.' }
```

Comment:
How about adding a test to your api_controller_spec.rb that checks for this?

Reply:
Done